src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -730,3 +709,56 @@ fn filename<'a, 'cfg>(cx: &mut Context<'a, 'cfg>, unit: &Unit<'a>) -> String {
     };
     format!("{}{}-{}", flavor, kind, file_stem)
 }
+
+/// Parses the dep-info file coming out of rustc into a Cargo-specific format.
+///
+/// This function will parse `rustc_dep_info` as a makefile-style dep info to
+/// learn about the all files which a crate depends on. This is then
+/// re-serialized into the `cargo_dep_info` path in a Cargo-specific format.
+///
+/// The `pkg_root` argument here is the absolute path to the directory
+/// containing `Cargo.toml` for this crate that was compiled. The paths listed
+/// in the rustc dep-info file may or may not be absolute but we'll want to
+/// consider all of them relative to the `root` specified.
+///
+/// The `rustc_cwd` argument is the absolute path to the cwd of the compiler
+/// when it was invoked.
+///
+/// The serialized Cargo format will contain a list of files, all of which are
+/// relative if they're under `root`. or absolute if they're elsewehre.
```

Comment:
Hm, weren't depinfo files, produced by Cargo, used by some other tools?  https://github.com/rust-lang/cargo/pull/3557

So, this change will break such clients :(

Reply:
Oh sure!

I think though (at least the intention) that these files are just those in `.fingerprint`, so the ones like `target/debug/foo.d` should still be present unmangled from rustc itself.